plugins/upload-assets-to-s3/src/tasks/upload-assets-to-s3.ts

```diff
@@ -64,9 +65,16 @@ export default class UploadAssetsToS3 extends Task<typeof UploadAssetsToS3Schema
             CacheControl: options.cacheControl
           }
           currentBucket = params.Bucket
-          const data = await s3.upload(params).promise()
-          this.logger.info(`Uploaded ${styles.filepath(filepath)} to ${styles.URL(data.Location)}`)
+          const data = s3.upload(params).promise()
+          uploadPromises.push(data)
```

Comment:
**suggestion**: we could rewrite this as a [`.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) over `bucketByEnv` rather than a for loop, which will save you creating an empty array and pushing to it each iteration cycle. something like
```ts
bucketByEnv.map((bucket) => {
  const params = {
     ... // snip
  }
  currentBucket = params.Bucket
  return s3.upload(params).promise()
})